airflow/gcp/operators/bigquery.py

```diff
@@ -242,9 +252,14 @@ class BigQueryGetDataOperator(BaseOperator):
     :type dataset_id: str
     :param table_id: The table ID of the requested table. (templated)
     :type table_id: str
+    :param start_index: (Optional) Start row index of the table.
+    :type start_index: str
     :param max_results: The maximum number of records (rows) to be fetched
         from the table. (templated)
-    :type max_results: str
+    :type max_results: int
+    :param page_token: (Optional) Page token, returned from a previous call,
```

Comment:
This is the implementation detail of the library. This should only be in the hook, and the hook method should always return the full range of data. Otherwise, It will not possible to migrate to different library eg. https://pypi.org/project/google-cloud-bigquery/ 

My team is working on migrating integration(if possible) on google-cloud-python. It's a rpart of recommendation of GCP integration guide. We have started retrofiting this operator also - https://github.com/PolideaInternal/airflow/issues/231

@TobKed Can you provide a detail information about retrofiting integration with BigQuery?

Reply:
IMO page token should should not be passed as a parameter to Operator, maybe in Hook but also I don't think it is good idea, as @mik-laj said: 

> the hook method should always return the full range of data

A method in `google-cloud-bigquery` accepts `page_token` parameter hence it may be optionally used in hook
https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html?highlight=list_rows#google.cloud.bigquery.client.Client.list_rows

WDYT @mik-laj @nuclearpinguin